### PR TITLE
Feature/paypal errors

### PR DESCRIPTION
**Pull request by ankitecd**

### CODE REVIEW

**Review by sagararyal — APPROVED** [review]

File: src/Donations/PaymentMethods/NewPaypal.tsx [hunk 1]
```diff
@@ -67,11 +67,28 @@ function NewPaypal({
   }
 
   const onError = (data) => {
-    setPaymentError(`Your order ${data.orderID} failed due to some error.`);
+    setPaymentError(`Your order failed due to some error.`);
```
Comment (norbertschuler) [inline src/Donations/PaymentMethods/NewPaypal.tsx:70]: If this is an information displayed to the user, maybe it should be translated?

File: src/Donations/PaymentMethods/NewPaypal.tsx [hunk 2]
```diff
@@ -67,11 +67,28 @@ function NewPaypal({
...
+      type: "sdk",
+      status: "error",
+      errorMessage: data?.message,
+    };
+    payDonationFunction("paypal", "paypal", data);
   };
 
-  const onCancel = (data) => {
+  const onCancel = (data, actions) => {
     setPaymentError("Order was cancelled, please try again");
```
Comment (norbertschuler) [inline src/Donations/PaymentMethods/NewPaypal.tsx:83]: If this is an information displayed to the user, maybe it should be translated?

File: src/Donations/PaymentMethods/NewPaypal.tsx [hunk 3]
```diff
@@ -67,11 +67,28 @@ function NewPaypal({
   }
 
   const onError = (data) => {
-    setPaymentError(`Your order ${data.orderID} failed due to some error.`);
+    setPaymentError(`Your order failed due to some error.`);
+
+    // This function shows a transaction success message to your buyer.
```
Comment (norbertschuler) [inline src/Donations/PaymentMethods/NewPaypal.tsx:72]: Why does the comment is talking about a `transaction success message` if something failed?

File: src/Donations/PaymentMethods/NewPaypal.tsx [hunk 4]
```diff
@@ -67,11 +67,28 @@ function NewPaypal({
...
+      errorMessage: data?.message,
+    };
+    payDonationFunction("paypal", "paypal", data);
   };
 
-  const onCancel = (data) => {
+  const onCancel = (data, actions) => {
     setPaymentError("Order was cancelled, please try again");
+
+    // This function shows a transaction success message to your buyer.
```
Comment (norbertschuler) [inline src/Donations/PaymentMethods/NewPaypal.tsx:85]: Why does the comment is talking about a `transaction success message` if something got cancelled?

**Review by norbertschuler — CHANGES_REQUESTED** [review]

**Pull request merged**